### PR TITLE
server: Add Mailbox.Close, make ListMailboxes return MailboxInfo

### LINKED ISSUE #337: Add Mailbox.Close
I am working on the maildir-based backend that uses BoltDB for caches and metadata.
BoltDB DB is created per mailbox. backend.Mailbox implementation opens it on demand, but it also needs to close it at some point. Currently Close() is missing for Mailbox so I cannot do it.

Proposal: If Mailbox implements io.Closer, call Close() after use. Not sure about ListMailboxes.

### PR DESCRIPTION
**Pull request by foxcpp**
~~This change is backward compatible with v1.~~

go-imap-unselect needs an update to avoid 'leaking' mailboxes, I propose to merge it into go-imap along the way per #322.

Closes #337.

### CODE REVIEW

**Comment by emersion** [comment]
Honestly I'm not a big fan of this change. This doesn't fit well with the v1 interfaces. I'd prefer to start working on a v2 branch to address this issue (by adding a mandatory `Mailbox.Close` method and making `ListMailboxes` return `MailboxInfo` structs directly).

**Comment by foxcpp** [comment]
Alright, I will update this PR this weekend (hopefully, soju :\ ).

**Comment by foxcpp** [comment]
I guess, https://github.com/foxcpp/go-imap-maildir will be built along with go-imap v2 then :)

File: server/cmd_auth.go [hunk 1]
```diff
@@ -110,6 +114,7 @@ func (cmd *Subscribe) Handle(conn Conn) error {
 	if err != nil {
 		return err
 	}
+	defer mbox.Close()
 
 	return mbox.SetSubscribed(true)
```
Comment (emersion) [inline server/cmd_auth.go:119]: Maybe this should be a `User` method instead, so that a mailbox is only opened on SELECT?

**Comment by emersion** [comment]
Overall looks good. We could address the comment in another PR if you prefer.

File: server/cmd_auth.go (hunk 1 above)
Comment (foxcpp) [reply server/cmd_auth.go:119]: I agree. I think I even mentioned that somewhere but this is out-of-scope of this PR.

**Pull request merged**